alchemy/alchemy.py

```diff
@@ -980,31 +1054,87 @@ def _alchemicallyModifyNonbondedForce(self, system, reference_force):
+            if self.annihilate_electrostatics and ((iatom in alchemical_atom_indices) or (jatom in alchemical_atom_indices)):
+                # If exception (and not exclusion), add special CustomBondForce term to handle alchemically-modified electrostatics exclusion
+                if abs(chargeprod.value_in_unit_system(unit.md_unit_system)) > 0.0:
+                    electrostatics_custom_bond_force.addBond(iatom, jatom, [chargeprod, sigma])
 
-        # TODO: Add back NonbondedForce terms for alchemical system needed in case of decoupling electrostatics or sterics via second CustomBondForce.
-        # TODO: Also need to change current CustomBondForce to not alchemically disappearing system.
+        # Create atom groups.
+        natoms = system.getNumParticles()
+        # Create sets
+        alchemical_atomset = set(alchemical_atom_indices) # only alchemically-modified atoms
+        all_atomset = set(range(system.getNumParticles())) # all atoms, including alchemical region
+        nonalchemical_atomset = all_atomset.difference(alchemical_atomset)
+        # Create lists
+        alchemical_atomlist = list(alchemical_atomset)
+        all_atomlist = list(all_atomset)
+        nonalchemical_atomlist = list(nonalchemical_atomset)
+
+        # Add back intra-alchemical sterics for decoupled sterics or electrostatics via CustomBondForce terms.
+        # TODO: Is there a better way to do this?
+        for iatom in alchemical_atomlist:
+            [charge1, sigma1, epsilon1] = nonbonded_force.getParticleParameters(iatom)
+            for jatom in alchemical_atomlist:
+                [charge2, sigma2, epsilon2] = nonbonded_force.getParticleParameters(jatom)
+                if (iatom < jatom) and ((iatom,jatom) not in exception_pairs):
+                    # Combining rules.
+                    # TODO: Generalize these for other mixing rules.
+                    chargeprod = charge1*charge2
+                    epsilon = unit.sqrt(epsilon1*epsilon2)
+                    sigma = 0.5*(sigma1 + sigma2)
+                    # Add decoupled terms.
+                    if (not self.annihilate_sterics) and (epsilon.value_in_unit_system(unit.md_unit_system) != 0):
+                        sterics_custom_bond_force.addBond(iatom, jatom, [sigma, epsilon])
+                    if (not self.annihilate_electrostatics) and (chargeprod.value_in_unit_system(unit.md_unit_system) != 0):
+                        electrostatics_custom_bond_force.addBond(iatom, jatom, [chargeprod, sigma])
+
+        # Turn off Lennard-Jones contribution from alchemically-modified particles.
+        for particle_index in range(nonbonded_force.getNumParticles()):
+            # Retrieve parameters.
+            [charge, sigma, epsilon] = nonbonded_force.getParticleParameters(particle_index)
+            if particle_index in alchemical_atom_indices:
+                nonbonded_force.setParticleParameters(particle_index, abs(0*charge), sigma, abs(0*epsilon))
```

Comment:
The `NonbondedForce` object must have the same number of particles as the `System`, also, deleting a particle would result in shifting all the indices around. OpenMM is smart enough that if charge=0 or epsilon=0 it ignores it in the electrostatic and Lennard-Jones calculations respectively.